### PR TITLE
[CoSim] several minor improvements

### PR DESCRIPTION
**Pull request by philbucher**
this PR contains several small improvements that came up in the recent past
mostly new checks were added

### CODE REVIEW

**Comment by philbucher** [comment]
Ping @adityaghantasala

File: applications/CoSimulationApplication/python_scripts/data_transfer_operators/copy.py [hunk 1]
```diff
@@ -19,13 +21,13 @@ def TransferData(self, from_solver_data, to_solver_data, transfer_options):
 
         transfer_options_list = transfer_options.GetStringArray()
 
+        # the order is IMPORTANT here!
```
Comment (adityaghantasala) [inline applications/CoSimulationApplication/python_scripts/data_transfer_operators/copy.py:24]: Yes .. is this order shown to user at some point ?

File: applications/CoSimulationApplication/python_scripts/data_transfer_operators/kratos_mapping.py [hunk 2]
```diff
@@ -32,30 +38,40 @@ def TransferData(self, from_solver_data, to_solver_data, transfer_options):
...
         variable_origin        = from_solver_data.variable
+        identifier_origin      = from_solver_data.solver_name + "." + model_part_origin_name
 
-        model_part_destinatinon      = to_solver_data.GetModelPart()
-        model_part_destinatinon_name = model_part_destinatinon.Name
-        variable_destination         = to_solver_data.variable
+        model_part_destination      = to_solver_data.GetModelPart()
+        model_part_destination_name = to_solver_data.model_part_name
+        variable_destination        = to_solver_data.variable
+        identifier_destination      = to_solver_data.solver_name + "." + model_part_destination_name
```
Comment (adityaghantasala) [inline applications/CoSimulationApplication/python_scripts/data_transfer_operators/kratos_mapping.py:48]: Nice !

**Review by adityaghantasala — APPROVED** [review]

File: applications/CoSimulationApplication/python_scripts/data_transfer_operators/copy.py (hunk 1 above)
Comment (philbucher) [reply applications/CoSimulationApplication/python_scripts/data_transfer_operators/copy.py:24]: this is meant in terms of implementation, for the user it does not matter

I will add an explanatory comment in the next PR

**Pull request merged**